src/seedu/addressbook/data/person/Address.java

```diff
@@ -50,13 +50,14 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
      * Returns true if a given string is a valid person email.
      */
     public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+        return test.matches(ADDRESS_VALIDATION_REGEX)
+        		&& (splitAddress(test).length == 4);
     }
 
```

Comment:
You might like to add further regex validations on the block street, postal code and unit.